### PR TITLE
Introduce ReplicaIDs as described in #1865.

### PR DESCRIPTION
**Pull request by bdarnell**
The replica IDs are stored and propagated through replica updates,
but not yet used.

### CODE REVIEW

File: proto/metadata.go [hunk 1]
```diff
@@ -140,6 +145,25 @@ func (r *RangeDescriptor) FindReplica(storeID StoreID) (int, *Replica) {
...
 
+// Validate performs some basic validation of the contents of a range descriptor.
+func (r *RangeDescriptor) Validate() error {
+	if r.NextReplicaID == 0 {
+		return util.Errorf("NextReplicaID must be non-zero")
+	}
+	seen := map[ReplicaID]struct{}{}
+	for _, rep := range r.Replicas {
+		if _, ok := seen[rep.ReplicaID]; ok {
+			return util.Errorf("ReplicaID %d was reused", rep.ReplicaID)
```
Comment (tamird) [inline proto/metadata.go]: this error uses a different tense than the one below

**Comment by tamird** [comment]
Heh, this is smaller than it looks. LGTM

**Comment by BramGruneir** [comment]
LGTM

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**